File: matlab/examples/rest_weighings.m
one = One();
%%

subject = '437';
wei = one.alyx_client.get(['/weighings?nickname=' subject]);

% wei = one.alyx_client.get(['/weighings']);
wei.date_time = time.jsonrest2serial(wei.date_time);
[~,ordre] = sort(wei.date_time );
wei = sel_struct(wei, ordre);


%%
figure,plot(wei.date_time, wei.weight,'.')


%%
wa = one.alyx_client.get(['/water-administrations']);
wa = one.alyx_client.get(['/water-administrations?nickname=' subject]);
wa.date_time = time.jsonrest2serial(wa.date_time);

File: python/oneibl/examples/rest_weighings.py
import matplotlib.pyplot as plt
import pandas as pd

from oneibl.one import ONE
from ibllib.time import isostr2date

one = ONE()

# this gets all weighings for all subjects
wei = one.alyx.rest('weighings', 'list')

# this gets the weighings for one subject
subject = '437'
wei = pd.DataFrame(one.alyx.rest('weighings', 'list', '?nickname=' + subject))
plt.plot_date(wei['date_time'].apply(isostr2date), wei['weight'])

# to list administrations for one subject, it is better to use the subjects endpoint
sub_info = one.alyx.rest('subjects', 'read', '437')
wei = pd.DataFrame(sub_info['weighings'])
wei['date_time'].apply(isostr2date)
wei.sort_values('date_time', inplace=True)
plt.plot(wei.date_time, wei.weight)
